### PR TITLE
[Drop-In UI] Moved CameraModeButtonComponent outside Drop-In UI

### PR DESCRIPTION
**Pull request by tomaszrybakiewicz**
Part of https://github.com/mapbox/navigation-sdks/issues/1780

### TL;DR

This PR adds two new ComponentInstallers: `navigationCamera()` that simplifies integration of `NavigationCamera` and `cameraModeButton()` that toggles between NavigationCameraState.FOLLOWING and NavigationCameraState.OVERVIEW  on button press.

```kotlin
MapboxNavigationApp.installComponents(/* LifecycleOwner */ this) {
    cameraModeButton(binding.cameraModeButton)
    navigationCamera(binding.mapView)
}
```

### Description

- moved `CameraModeButtonComponent` to `libnavui-maps` module
- decoupled `CameraModeButtonComponent` from the `Store`by introducing `CameraModeButtonComponentContract`
- added both default and dropin-ui `CameraModeButtonComponentContract` implementations
- introduced `NavigationCameraComponent` that simplifies `MapboxNavigationViewportDataSource` and `NavigaitonCamera` integration
- introduced `NavigationCameraGestureComponent` that registers `NavigationBasicGesturesHandler` when attached (NavigationBasicGesturesHandler updates NavigationCamera to IDLE when map gesture is detected)

### CODE REVIEW

**Review by abhishek1508 — APPROVED** [review]
LGTM. `static analysis` and `unit-tests-core` are failing.

**Comment by tomaszrybakiewicz** [comment]
@abhishek1508 I fixed the `static-analysis` problems. Not sure why the `changelog-verification` check is failing now. CHANGELOG does have the entries that reference this PR.

**Comment by dzinad** [comment]
> Not sure why the changelog-verification check is failing now. CHANGELOG does have the entries that reference this PR.

Exception: The changelog entry "" should contain a link to the original PR that matches `\[#\d+]\(https:\/\/github\.com\/******\/******-navigation-android\/pull\/\d+\)`

**Comment by dzinad** [comment]
> Not sure why the changelog-verification check is failing now. CHANGELOG does have the entries that reference this PR.

I think it does not like the empty line... Will fix.

**Comment by abhishek1508** [comment]
> @abhishek1508 I fixed the `static-analysis` problems. Not sure why the `changelog-verification` check is failing now. CHANGELOG does have the entries that reference this PR.

We are seeing similar issues in android-auto and [ticket](https://github.com/mapbox/mapbox-navigation-android/issues/6220) has been created. Let us merge this PR. @tomaszrybakiewicz
